src/ResourceManager/Insights/Commands.Insights/MonitorCmdletBase.cs

```diff
@@ -0,0 +1,108 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Globalization;
+using System.Management.Automation;
+using System.Net;
+using Microsoft.Azure.Commands.ResourceManager.Common;
+using System;
+using Microsoft.Rest;
+using Microsoft.Rest.Azure;
+
+namespace Microsoft.Azure.Commands.Insights
+{
+    /// <summary>
+    /// Base class for the Azure Insights SDK Cmdlets
+    /// </summary>
+    public abstract class MonitorCmdletBase : AzureRMCmdlet
+    {
+        /// <summary>
+        /// Executes the Cmdlet. This is a callback function to simplify the exception handling
+        /// </summary>
+        protected abstract void ProcessRecordInternal();
+
+        /// <summary>
+        /// Execute the cmdlet
+        /// </summary>
+        public override void ExecuteCmdlet()
+        {
+            try
+            {
+                this.ProcessRecordInternal();
+            }
+            catch (AggregateException ex)
+            {
+                // Process the exception to be as informative as possible to the user
+                var exTemp = ex.Flatten().InnerException ?? ex;
+                string exName = exTemp.GetType().Name;
+                string message = exTemp.Message;
+                string code = null;
+                HttpStatusCode? statusCode = null;
+                string reasonPhrase = null;
+
+                if (exTemp is RestException)
+                {
+                    // All the following Exceptions have the same structure, but their common ancestor does not contain Body nor Response
+                    var cloudException = exTemp as CloudException;
+                    if (cloudException != null)
```

Comment:
Since we we are generating two clients the ErrorResponseException is present in two namespaces. So I need one if for each type.

Then I am using variables the I assigned inside each if to use the values in the throw -last line of the catch -.